src/libs/actions/Report.js

```diff
@@ -1201,6 +1133,108 @@ Onyx.connect({
+        [sequenceNumber]: {
+            pendingAction: CONST.RED_BRICK_ROAD_PENDING_ACTION.DELETE,
+        },
+    };
+    const optimisticReport = {
+        lastMessageText: ReportActions.getLastVisibleMessageText(reportID, {
+            [sequenceNumber]: {
+                ...reportAction,
+            },
+        }),
+    };
+
+    // If the deleted comment is more recent than our last read comment, update the unread count
+    if (sequenceNumber > getLastReadSequenceNumber(reportID)) {
+        optimisticReport.unreadActionCount = Math.max(getUnreadActionCount(reportID) - 1, 0);
+    }
+
+    // List of optmistic data to change
+    const optimisticData = [
+        {
+            onyxMethod: CONST.ONYX.METHOD.MERGE,
+            key: `${ONYXKEYS.COLLECTION.REPORT_ACTIONS}${reportID}`,
+            value: optimisticReportActions,
+        },
+        {
+            onyxMethod: CONST.ONYX.METHOD.MERGE,
+            key: `${ONYXKEYS.COLLECTION.REPORT}${reportID}`,
+            value: optimisticReport,
+        },
+    ];
+
+    // On Success remove the message content and pendingAction state
+    const successData = {
+        [sequenceNumber]: {
+            pendingAction: null,
+            message: [{
+                html: '',
+                text: '',
+            }],
+        },
+    };
+
+    // On Error clear the pendingAction state and revert the message
+    const failureData = [
+        {
+            onyxMethod: CONST.ONYX.METHOD.MERGE,
+            key: `${ONYXKEYS.COLLECTION.REPORT_ACTIONS}${reportID}`,
+            value: {
+                [sequenceNumber]: {
+                    message: reportAction.message,
+                    pendingAction: null,
+                },
+            },
+        },
+
+        // {
+        //     onyxMethod: CONST.ONYX.METHOD.MERGE,
+        //     key: `${ONYXKEYS.COLLECTION.REPORT}${reportID}`,
+        //     value: {
+        //         lastMessageText:
+        //         unreadActionCount:
+        //     }
+        // }
+    ];
+
+    const parameters = {
+        reportID,
+        clientID: reportAction.clientID,
+        sequenceNumber,
+        reportActionID: reportAction.reportActionID,
+    };
+    API.write('DeleteComment', parameters, {optimisticData, successData, failureData});
+
+    // If all fails Reverse Optimistic Response:
+    // reportActionsToMerge[sequenceNumber] = {
+    //     ...reportAction,
+    //     message: oldMessage,
+    // };
+
+    // if (sequenceNumber > getLastReadSequenceNumber(reportID)) {
+    //     Onyx.merge(`${ONYXKEYS.COLLECTION.REPORT}${reportID}`, {
+    //         unreadActionCount: getUnreadActionCount(reportID) + 1,
+    //     });
+    // }
+
+    // Onyx.merge(`${ONYXKEYS.COLLECTION.REPORT}${reportID}`, {
+    //     lastMessageText: ReportActions.getLastVisibleMessageText(reportID, reportActionsToMerge),
+    // });
+
+    // Onyx.merge(`${ONYXKEYS.COLLECTION.REPORT_ACTIONS}${reportID}`, reportActionsToMerge);
```

Comment:
Is this getting removed?

Reply:
Yes! I'm keeping it here to check what the method did before but I'll remove it